lib/dotloop/loop.rb

```diff
@@ -2,38 +2,105 @@ module Dotloop
   class Loop
     attr_accessor :client
     BATCH_SIZE = 50
-    # attr_accessor :profile_id
-    # attr_accessor :batch_number
-    # attr_accessor :batch_size
-    # attr_accessor :status_ids
-    # attr_accessor :compliance_status_ids
-    # attr_accessor :tag_ids
-    # attr_accessor :sort_by
-    # attr_accessor :search_query
-    # attr_accessor :tag_names
-    # attr_accessor :created_by_me
+    MAX_LOOPS = 500
 
     def initialize(client:)
       @client = client
     end
 
-    def all(profile_id:)
-      batches = []
-      batch_number = 1
-      loop do
-        batch = retrieve_batch(profile_id: profile_id, batch_size: BATCH_SIZE, batch_number: batch_number)
-        batches += batch
-        batch_number += 1
-        break if batch.size < BATCH_SIZE
+    # rubocop:disable Metrics/ParameterLists
+    def all(
+      profile_id:, batch_number: nil, batch_size: nil, status_ids: [],
```

Comment:
again, if we rewrite this as:

```
def all(*options)
    options = query_params(options)
    # after this we now have a validated and default hash of all the options
    # ...
end
```

Because the BATCH_SIZE is nicely defaulted in the query_params method.